pkg/controllers/provisioningv2/cluster/remove.go

```diff
@@ -130,3 +133,17 @@ func (h *handler) doClusterRemove(cluster *v1.Cluster) func() (string, error) {
 		return "", h.kubeconfigManager.DeleteUser(cluster.Namespace, cluster.Name)
 	}
 }
+
+func capiClusterIsRancherOwned(cluster *v1.Cluster, capiCluster *capi.Cluster) bool {
```

Comment:
is there a possibility that any argument will be nil?

Reply:
No, it would've failed before it got here.